view_info.hh

```diff
@@ -58,6 +58,10 @@ public:
         return _raw.where_clause();
     }
 
+    /*!
+     * \brief returns true if this view is backing a secondary index
+     */
+    bool is_secondary_index() const;
```

Comment:
This is a layering violation, a materialized view shouldn't know about different things it is used for as an infrastructure. In fact we used to have this method and @psarna removed it in an effort to fix this layering violation.

Reply:
so you remove the method, but still do that in the code,
see the code in: cql3::statements::select_statement& view_info::select_statement()

I'll place the logic somewhere else

Reply:
The code in `view_info::select_statement` is a necessary evil due to backward compatibility, it's marked with this comment:
```cpp
// FIXME(sarna): legacy code, should be removed after "computed_columns" feature is guaranteed
// to be available on every node. Then, we won't need to check if this view is backing a secondary index.
```
It would be best to avoid checking if if the view backs an index in the view code itself, but instead on an upper layer (e.g. a code that iterates over all views of a table could exclude ones backed by an index, because a table knows its indexes and their accompanying views)

Reply:
@psarna you're right but that is not the case I'm facing, I'm writing a describe method in a schema (similar to a cql DESCRIBE command) and if the schema in hand is a view I need to know how to treat it.

Anyway, as I replied to @denesb I'll move the same hack that is used here to somewhere else for my need.